src/components/images/imageLoader.js

```diff
@@ -11,22 +12,75 @@ import 'css!./style';
         fillImageElement(elem, source);
     }
 
+    async function itemBlurhashing(target, blurhashstr) {
+        if (blurhash.isBlurhashValid(blurhashstr)) {
+            // Although the default values provided by blurhash devs is 32x32, 18x18 seems to be the sweetest spot possible,
+            // cramping up a lot the performance and reducing the memory usage, while retaining almost full blur quality.
+            // Lower values had more visible pixelation
+            let width = 18;
+            let height = 18;
+            let pixels;
+            try {
+                pixels = blurhash.decode(blurhashstr, width, height);
+            } catch (err) {
+                console.error('Blurhash decode error: ', err);
+                target.classList.add('non-blurhashable');
+                return;
+            }
+            let canvas = document.createElement('canvas');
+            canvas.width = width;
+            canvas.height = height;
+            let ctx = canvas.getContext('2d');
+            let imgData = ctx.createImageData(width, height);
+
+            imgData.data.set(pixels);
+            ctx.putImageData(imgData, 0, 0);
+
+            let child = target.appendChild(canvas);
+            child.classList.add('blurhash-canvas');
+            child.style.opacity = 1;
+            if (userSettings.enableFastFadein()) {
+                child.classList.add('lazy-blurhash-fadein-fast');
+            } else {
+                child.classList.add('lazy-blurhash-fadein');
+            }
+
+            target.classList.add('blurhashed');
+            target.removeAttribute('data-blurhash');
+        }
+    }
+
+    function switchCanvas(elem) {
+        let child = elem.getElementsByClassName('blurhash-canvas')[0];
+        if (child) {
+            child.style.opacity = elem.getAttribute('data-src') ? 1 : 0;
+        }
+    }
+
     export function fillImage(entry) {
         if (!entry) {
             throw new Error('entry cannot be null');
         }
-
+        let target = entry.target;
         var source = undefined;
-        if (entry.target) {
-            source = entry.target.getAttribute('data-src');
+
+        if (target) {
+            source = target.getAttribute('data-src');
+            var blurhashstr = target.getAttribute('data-blurhash');
         } else {
             source = entry;
         }
 
+        if (!target.classList.contains('blurhashed', 'non-blurhashable') && userSettings.enableBlurhash() && blurhashstr) {
+            itemBlurhashing(target, blurhashstr);
+        } else if (userSettings.enableBlurhash() && !blurhashstr && !target.classList.contains('blurhashed')) {
+            target.classList.add('non-blurhashable');
+        }
```

Comment:
@dmitrylyzo Now I don't remember why I put that there, but let me do my testing checklist and let's see if everything it's fine without it :P

Reply:
@dmitrylyzo After testing, now I remember why I did that! Items that don't have a blurhash string will get the non-blurhashable class when the blurhash option in usersettings is disabled.

Behavioral-wise, removing this doesn't make any difference. I just added it because, in my mind, it didn't make any sense to apply a class that was supposed to specify that we can't use the blurhash feature when the blurhash feature is disabled :).

Now I don't know what to do with this! :P